zebra-chain/src/primitives/zcash_primitives.rs

```diff
@@ -29,11 +29,11 @@ impl zp_tx::components::transparent::Authorization for TransparentAuth<'_> {
 // In this block we convert our Output to a librustzcash to TxOut.
 // (We could do the serialize/deserialize route but it's simple enough to convert manually)
 impl zp_tx::sighash::TransparentAuthorizingContext for TransparentAuth<'_> {
-    fn input_amounts(&self) -> Vec<zp_tx::components::amount::Amount> {
+    fn input_amounts(&self) -> Vec<zp_tx::components::amount::NonNegativeAmount> {
         self.all_prev_outputs
             .iter()
             .map(|prevout| {
-                zp_tx::components::amount::Amount::from_nonnegative_i64_le_bytes(
+                zp_tx::components::amount::NonNegativeAmount::from_nonnegative_i64_le_bytes(
```

Comment:
We need to convert here from zebra `Amount<NonNegative>` type to a zcash_primitives `NonNegativeAmount` here. I think `from_nonnegative_i64_le_bytes` is good enough here as zebra `Amount` is `i64`.